### PR TITLE
refactor: replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`

### PR DESCRIPTION
**Pull request by Juneezee**
The package `github.com/ghodss/yaml` is no longer actively maintained. See discussion in https://github.com/ghodss/yaml/issues/75 and https://github.com/ghodss/yaml/issues/80. [`sigs.k8s.io/yaml`](https://github.com/kubernetes-sigs/yaml) is a permanent fork of `github.com/ghodss/yaml`.

The notable change is that `github.com/ghodss/yaml` uses `gopkg.in/yaml.v2 v2.2.2`, but `sigs.k8s.io/yaml` uses `gopkg.in/yaml.v2 v2.4.0`. Changes can be seen here [v2.2.2...v2.4.0](https://github.com/go-yaml/yaml/compare/v2.2.2...v2.4.0), mostly bug fixes.

### CODE REVIEW

**Comment by r4f4** [comment]
Is there some place left still using `github.com/ghodss/yaml`? I'd expect vendoring changes to remove that dep.

File: pkg/asset/installconfig/installconfigbase.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (a *AssetBase) LoadFromFile(f asset.FileFetcher) (found bool, err error) {
 		err = errors.Wrapf(err, "failed to parse first occurrence of unknown field")
 		logrus.Warnf(err.Error())
 		logrus.Info("Attempting to unmarshal while ignoring unknown keys because strict unmarshaling failed")
-		if err = yaml.UnmarshalStrict(file.Data, config); err != nil {
+		if err = yaml.Unmarshal(file.Data, config); err != nil {
```
[the remark this answers is not in the document]
Comment (rna-afk) [reply pkg/asset/installconfig/installconfigbase.go:56]: Does sigs not have UnmarshalStrict? We changed this to strict so the users know of fields they added to the install config that the installer does not recognize.

[the remark this answers is not in the document]
Comment (Juneezee) [reply pkg/asset/installconfig/installconfigbase.go:56]: `sigs.k8s.io/yaml` has `UnmarshalStrict`, but their `UnmarshlalStrict` disallows unknown fields automatically. See https://github.com/kubernetes-sigs/yaml/blob/65d71bb2ae7234d12629f43f77e1a44512057bba/yaml.go#L64

If we use `UnmarshalStrict` here we will get the error again.

**Comment by Juneezee** [comment]
> Is there some place left still using `github.com/ghodss/yaml`? I'd expect vendoring changes to remove that dep.

@r4f4 One of our dependency https://github.com/openshift/cluster-api-provider-libvirt is still using `github.com/ghodss/yaml`.

```
$ go mod why -m github.com/ghodss/yaml
# github.com/ghodss/yaml
github.com/openshift/installer/pkg/asset/cluster
github.com/openshift/cluster-api-provider-libvirt/pkg/apis/libvirtproviderconfig/v1beta1
github.com/ghodss/yaml
```

File: pkg/asset/installconfig/installconfigbase.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rna-afk) [reply pkg/asset/installconfig/installconfigbase.go:56]: Ah missed the UnmarshallStrict above, this is fine.

**Comment by r4f4** [comment]
> > Is there some place left still using `github.com/ghodss/yaml`? I'd expect vendoring changes to remove that dep.
> 
> @r4f4 One of our dependency https://github.com/openshift/cluster-api-provider-libvirt is still using `github.com/ghodss/yaml`.
> 
> ```
> $ go mod why -m github.com/ghodss/yaml
> # github.com/ghodss/yaml
> github.com/openshift/installer/pkg/asset/cluster
> github.com/openshift/cluster-api-provider-libvirt/pkg/apis/libvirtproviderconfig/v1beta1
> github.com/ghodss/yaml
> ```

So maybe we could use a replace directive to take care of that until cluster-api-provider-libvirt moves away from it?
```bash
replace github.com/ghodss/yaml => sigs.k8s.io/yaml
```

**Comment by Juneezee** [comment]
> So maybe we could use a replace directive to take care of that until cluster-api-provider-libvirt moves away from it?
> ```shell
> replace github.com/ghodss/yaml => sigs.k8s.io/yaml
> ```

@r4f4 Go doesn't seem to allow this

```
$ go mod tidy
go: sigs.k8s.io/yaml@v1.3.0 used for two different module paths (github.com/ghodss/yaml and sigs.k8s.io/yaml)
```

Is it okay if I go open a PR in https://github.com/openshift/cluster-api-provider-libvirt to replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`?

**Comment by r4f4** [comment]
> > So maybe we could use a replace directive to take care of that until cluster-api-provider-libvirt moves away from it?
> > ```shell
> > replace github.com/ghodss/yaml => sigs.k8s.io/yaml
> > ```
> 
> @r4f4 Go doesn't seem to allow this
> 
> ```
> $ go mod tidy
> go: sigs.k8s.io/yaml@v1.3.0 used for two different module paths (github.com/ghodss/yaml and sigs.k8s.io/yaml)
> ```

Oh I see.

> Is it okay if I go open a PR in https://github.com/openshift/cluster-api-provider-libvirt to replace `github.com/ghodss/yaml` with `sigs.k8s.io/yaml`?

Yep, I think that's the proper way to solve this problem.

**Comment by Juneezee** [comment]
Marking this PR as draft until https://github.com/openshift/cluster-api-provider-libvirt/pull/252 is merged.

**Review by r4f4 — COMMENTED** [review]
/approve

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Pull request merged**